### ISSUE TITLE
Renaming 'graalphp'

### ISSUE THREAD

**Issue by abertschi** [abertschi/graalphp#57]
We may want to reconsider the name 'graalphp'.  Other language implementations were asked to do so as well due to GraalVM trademark issues.

Suggestions:
```
AFAIK you are allowed to use “gr + <something>“, so “grPHP” would be ok (although probably not a good name). I think you could call it TrufflePHP but I am not sure anyone else is using it (apart from https://github.com/direktspeed/trufflephp).
```

**Comment by mukel** [comment]
Some implementations adopt names unrelated to Graal or Truffle e.g. Espresso (Java), Sulong (LLVM), FastR (R)... Naming is hard!

**Comment by abertschi** [comment]
Agree, naming is hard. I am open for suggestions. TrufflePHP seems to follow the pattern which other implementations adapted.

**Comment by frank-dspeed** [comment]
i think i named it trufflephp i also need to call the maintainers of the other jphp implementations so they can jump in
i think trufflephp is the most best name choice

**Comment by chrisseaton** [comment]
TrufflePHP seems good to me.

**Comment by MWGuy** [comment]
I think TrufflePHP is good name for this language implementation :eyes:

**Comment by frank-dspeed** [comment]
@abertschi i will merg your project into direktspeed/trufflephp it was made as placeholder as i have some people that want to jump into finishing the implementation that you did. I just started preperation to make that a good community project with all the needed stuff. We need a lot of documentation and github actions to publish pages with the current status and all that.

**Comment by abertschi** [comment]
> @abertschi i will merg your project into direktspeed/trufflephp it was made as placeholder as i have some people that want to jump into finishing the implementation that you did. I just started preperation to make that a good community project with all the needed stuff. We need a lot of documentation and github actions to publish pages with the current status and all that.

@frank-dspeed 
I have not decided a new name yet, but the name 'graalphp' cannot stay. If you are interested in contributing to the project, you are very welcome to fork the code. See license file in the root directory. I appreciate your commitment and initiative.

More name ideas:
- elephant
- littleElephant
- elephant in some other language than english

**Comment by chrisseaton** [comment]
A colleague of mine (can't remember who to credit them sorry) suggested PHastP.

**Comment by frank-dspeed** [comment]
@abertschi out of any view trufflephp is the best solution for the name at all because it can then get inferred that it is based on Java(GraalVM) and when some one searches he will search graalphp trufflephp but i am not sure if Oracle as a patent on Graal? i think everyone can call anything graal* if you got infos about why you could not use graalphp i would be interrested.

**Comment by abertschi** [comment]
> @abertschi out of any view trufflephp is the best solution for the name at all because it can then get inferred that it is based on Java(GraalVM) and when some one searches he will search graalphp trufflephp but i am not sure if Oracle as a patent on Graal? i think everyone can call anything graal* if you got infos about why you could not use graalphp i would be interrested.

Several people (at Oracle Labs and people involved in other language implementations) told me that 'graalphp' may lead to legal issues because the name GraalVM is intellectual property of Oracle.

**Comment by frank-dspeed** [comment]
@abertschi i do not think so ok they own graalvm but that does stop no one from using graalpm and you did a language implementation based on it. It is the Same like Calling a Application WinSimCity and then fear get law sued by Microsoft because they own Windows. I asked a German Lawyer about that he did lol and sayed there will nothing happen i do not know about other countrys but in the EU it is save.

**Comment by frank-dspeed** [comment]
@abertschi do not forget to do a pull request here and add it https://github.com/oracle/graal/blob/master/truffle/docs/Languages.md to the list of languages.

**Comment by abertschi** [comment]
I would like to pragmatic. I told them I will think of a new name. It's totally fine to rename it to trufflephp or another name. This is not very important to me. I like the 'AST' in PHastP but the name looks rather cryptic :)

**Comment by ivanjaros** [comment]
> PHastP

My thoguhts exactly...
![](https://i.imgflip.com/4g8nmy.jpg)

**Comment by frank-dspeed** [comment]
@abertschi i just invited to as maintainer for https://github.com/direktspeed/trufflephp and did merge and renamed the project to trufflephp i will now start to define some basic schedules and tell other people that they now can contribute to it, My Main focus will be on php4x a php engine that uses vertx for nativ stuff like mysql, net, http, all the php module stuff.

**Comment by abertschi** [comment]
> @abertschi i just invited to as maintainer for https://github.com/direktspeed/trufflephp and did merge and renamed the project to trufflephp i will now start to define some basic schedules and tell other people that they now can contribute to it, My Main focus will be on php4x a php engine that uses vertx for nativ stuff like mysql, net, http, all the php module stuff.

Can you elaborate more on this? What is php4x and your wider idea with vertx. Are you a developer? You say you want to contribute to make it quote a  "good community project with all the needed stuff". Can you elaborate on this and on your "basic schedule to tell other people that they now can contribute to it"?

What would be the reasoning in using the repository you listed: https://github.com/direktspeed/trufflephp. As far as I can see there are no contributions yet. I see it is a fork of some other work, "kishida/phpparser". How is it related to what I have done?

I think it would make much more sense to create a new github organization.  I myself have already three repositories (this is the main one with the language implementation) related to what I have done until this point. If you are interested in having some coordination role we can both be administrators of the github organization. And you can start with what you plan with vertx in a dedicated repository or feature branches within that organization.

https://github.com/trufflephp

In terms of future work there are still many tasks to work on:
- Language features (OO, traits, namespaces, reading multiple files, data types, exceptions, and more).
- Integration of PHP's native libraries. I see JPHP follows an approach which re-implements many libraries. I believe an approach
  which reuses and integrates existing libraries is more viable. This depends on how efficiently these libraries can be integrated and needs to be further evaluated. I see @chrisseaton blogged about what they have done with truffleruby some years ago: https://chrisseaton.com/truffleruby/cext/. This sounds very interesting. This https://github.com/php/php-src/blob/master/ext/standard/basic_functions.c is the standard library of PHP. It is coupled with Zend Engine internals to for instance read parameter off their parameter data structures `ZEND_PARSE_PARAMETERS_START`.
- Web server integration

Renaming this project to a new name will involve changing package names and maven artifacts.
I need to check if I can easily rename the github repository in order not to break existing hyperlinks.

**Comment by abertschi** [comment]
I talked with some folks at Oracle Labs and there is currently no strategic interest in officially supporting PHP.

**Comment by frank-dspeed** [comment]
@abertschi simply lets call or have a meeting about php4x vertx is a big async framework for java and php4vertx aka php4x is a Java Runtime that supplys Java Nativ Objects like FileSystem Network to graalphp/trufflephp. 

To Make a long Story Short you can see vertx and php4x as the new ecosystem of trufflephp that offers mysql and all the nativ other stuff.

**Comment by frank-dspeed** [comment]
here is a good example of it that is the web stack and it got done for ruby and javascript already we have generators for that and will add trufflephp

https://vertx.io/docs/#web

**Comment by solidpulse** [comment]
> More name ideas:
> 
> * elephant
> * littleElephant
> * elephant in some other language than english

@abertschi 

I propose "**Mammoth**", 

something close to an elephant and everyone understands the word, and why is it related to PHP(Elephant)

![Mammoth Logo](https://i.imgur.com/GHRzYeR.png)

I suppose Truffle and Graal are Oracle product names.
